akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -437,15 +437,15 @@ abstract class ShardCoordinator(typeName: String, settings: ClusterShardingSetti
       if (isMember(region)) {
         log.debug("ShardRegion registered: [{}]", region)
         aliveRegions += region
-        if (state.regions.contains(region))
+        if (state.regions.contains(region)) {
           region ! RegisterAck(self)
-        else {
+          allocateShardHomesForRememberEntities()
```

Comment:
@drewhk This was the missing call in the other commit. That will send a `GetShardHome` message to self.
`GetShardHome` is ignored until `hasAllRegionsRegistered()` and that is based on `aliveRegions` that is updated a few lines above. Yeah, rather subtle oversight. It fixes itself if there is another `GetShardHome` later but otherwise it would stall.